src/app/cli/src/commands/verify_command.rs

```diff
@@ -9,23 +9,29 @@
 
 use std::sync::Arc;
 
+use futures::TryStreamExt;
 use kamu::domain::*;
 use opendatafabric::*;
 
 use super::{BatchError, CLIError, Command};
 use crate::output::OutputConfig;
 use crate::VerificationMultiProgress;
 
-type GenericVerificationResult =
-    Result<Vec<(DatasetRef, Result<VerificationResult, VerificationError>)>, CLIError>;
+type GenericVerificationResult = Result<
+    Vec<(
+        DatasetRef,
+        Arc<Result<VerificationResult, VerificationError>>,
```

Comment:
Why the result requires `Arc`? I think it can be avoided. We never access it from multiple threads, just transfer or copy the object

Reply:
There was small issue with error handling in verify_command, and it was some kind of temporary solution. Removed